src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/TestEnvironment.cs

```diff
@@ -87,32 +83,103 @@ public static VisualStudioInstance GetVisualStudioInstance ()
+					}
 				}
 			}
 		}
 
-		static string _dotNetAndroidSdkDirectory;
-		public static string DotNetAndroidSdkDirectory {
+		public static string DotNetPreviewDirectory => Path.Combine (XABuildPaths.PrefixDirectory, "dotnet");
+
+		public static string DotNetPreviewPacksDirectory => UseLocalBuildOutput ? LocalPacksDir : DefaultPacksDir;
+
+		public static string DotNetPreviewAndroidSdkDirectory => UseLocalBuildOutput ? LocalDotNetAndroidSdkDirectory : DefaultDotNetAndroidSdkDirectory;
+
+		public static string WorkloadPackOverridePath => Path.Combine (XABuildPaths.PrefixDirectory, "lib");
+
+		public static string WorkloadManifestOverridePath => Path.Combine (WorkloadPackOverridePath, "sdk-manifests");
+
+		static string DefaultPacksDir => Path.Combine (DotNetPreviewDirectory, "packs");
+
+		static string LocalPacksDir => Path.Combine (WorkloadPackOverridePath, "packs");
+
+		static string _defaultDotNetAndroidSdkDirectory;
+		static string DefaultDotNetAndroidSdkDirectory {
 			get {
-				if (!string.IsNullOrEmpty (_dotNetAndroidSdkDirectory)) {
-					return _dotNetAndroidSdkDirectory;
+				if (!string.IsNullOrEmpty (_defaultDotNetAndroidSdkDirectory)) {
+					return _defaultDotNetAndroidSdkDirectory;
 				}
-				var sdkName = IsMacOS ? "Microsoft.Android.Sdk.Darwin" :
-					IsWindows ? "Microsoft.Android.Sdk.Windows" :
-					"Microsoft.Android.Sdk.Linux";
-
-				var directories = from d in Directory.GetDirectories (Path.Combine (AndroidSdkResolver.GetDotNetPreviewPath (), "packs", sdkName))
-								  let version = ParseVersion (d)
-								  orderby version descending
-								  select d;
-				return _dotNetAndroidSdkDirectory = directories.FirstOrDefault ();
+				return _defaultDotNetAndroidSdkDirectory = GetDotNetAndroidSdkDir (DefaultPacksDir);
+			}
+		}
+
+		static string _localDotNetAndroidSdkDirectory;
+		static string LocalDotNetAndroidSdkDirectory {
+			get {
+				if (!string.IsNullOrEmpty (_localDotNetAndroidSdkDirectory)) {
+					return _localDotNetAndroidSdkDirectory;
+				}
+				return _localDotNetAndroidSdkDirectory = GetDotNetAndroidSdkDir (LocalPacksDir);
+			}
+		}
+
+		static string GetDotNetAndroidSdkDir (string packsDirectory)
+		{
+			var sdkName = IsMacOS ? "Microsoft.Android.Sdk.Darwin" :
+				IsWindows ? "Microsoft.Android.Sdk.Windows" :
+				"Microsoft.Android.Sdk.Linux";
+
+			var sdkDir = Path.Combine (packsDirectory, sdkName);
+			if (!Directory.Exists (sdkDir))
+				return string.Empty;
+
+			var dirs = from d in Directory.GetDirectories (sdkDir)
+				   let version = ParseVersion (d)
+				   orderby version descending
+				   select d;
+
+			return dirs.FirstOrDefault ();
+		}
+
+		public static bool UseLocalBuildOutput {
+			get {
+				var msbuildDir = Builder.UseDotNet ? Path.Combine (LocalDotNetAndroidSdkDirectory, "tools") : LocalMonoAndroidToolsDirectory;
+				return Directory.Exists (msbuildDir) && File.Exists (Path.Combine (msbuildDir, "Xamarin.Android.Build.Tasks.dll"));
+			}
+		}
```

Comment:
Is the only case this would normally be `false` when running on CI? The test lanes don't have a local build, they just have our packs installed?

Is it worth adding a `<summary>` mentioning this?

Reply:
That is correct, this should only be false for tests running on CI. I updated the description and added a comment for this.